### PR TITLE
Fixed order of energy charged for spawns.

### PR DESCRIPTION
**Pull request by ButAds**

### CODE REVIEW

File: src/processor/intents/spawns/_charge-energy.js [hunk 1]
```diff
@@ -15,15 +15,19 @@ module.exports = function(spawn, roomObjects, cost, bulk, roomController) {
...
+    spawns.sort(utils.comparatorDistance(spawn));
 
     spawns.forEach((i) => {
         var neededEnergy = Math.min(cost, i.energy);
         i.energy -= neededEnergy;
         cost -= neededEnergy;
         bulk.update(i, {energy: i.energy});
     });
+    if(cost <= 0) {
+        return;
```
Comment (PostCrafter) [inline src/processor/intents/spawns/_charge-energy.js]: This should be `return true;`.

File: src/processor/intents/spawns/_charge-energy.js [hunk 2]
```diff
@@ -15,15 +15,19 @@ module.exports = function(spawn, roomObjects, cost, bulk, roomController) {
...
-    extensions.sort(utils.comparatorDistance(spawn));
+    spawns.sort(utils.comparatorDistance(spawn));
 
     spawns.forEach((i) => {
         var neededEnergy = Math.min(cost, i.energy);
         i.energy -= neededEnergy;
         cost -= neededEnergy;
         bulk.update(i, {energy: i.energy});
     });
+    if(cost <= 0) {
```
Comment (PostCrafter) [inline src/processor/intents/spawns/_charge-energy.js]: I suggest adding an empty line above this one, to keep the style consistent.

File: src/processor/intents/spawns/_charge-energy.js [hunk 3]
```diff
@@ -15,15 +15,19 @@ module.exports = function(spawn, roomObjects, cost, bulk, roomController) {
...
+        return;
+    }
 
+    extensions.sort(utils.comparatorDistance(spawn));
```
Comment (PostCrafter) [inline src/processor/intents/spawns/_charge-energy.js]: I suggest adding an empty line below this one, to keep the style consistent.

**Comment by ButAds** [comment]
@PostCrafter Fixed the changes you suggested

**Pull request merged**